src/features/postSlice.js

```diff
@@ -0,0 +1,140 @@
+import { createAsyncThunk, createSlice } from "@reduxjs/toolkit"
+import { addPostService, deleteUserPostService, editPostService, getAllPostService, getUserPostService } from "../Services/postService";
+
+const initialState={
+    allPosts:[],
+    userPosts:[],
+    postToBeEdited:{}
+}
+export const setPostToBeEdited=createAsyncThunk("post/postToBeEdited",async(post,thunkAPI)=>{
+    return {...post}
+})
+
+export const getAllPosts=createAsyncThunk("post/getAllPosts",async(_,thunkAPI)=>{
+    try{
+        const response=await getAllPostService();
+        return response.data
+    }
+    catch(error){
+        return thunkAPI.rejectWithValue(error)
+    }
+})
+
+export const getUserPost=createAsyncThunk("post/getUserPost",async(username,thunkAPI)=>{
+    try{
+        const response=await getUserPostService(username)
+        console.log(response)
+    return response.data.posts
+
+    }catch(error){
+        return thunkAPI.rejectWithValue(error)
+    }
+})
+
+export const editUserPost=createAsyncThunk(
+    "post/editUserPost",async(postData,thunkAPI)=>{
+        try{
+            const token=localStorage.getItem("token")
+            const response=await editPostService(postData,token)
+            return response.data
+        }catch(error){
+            return thunkAPI.rejectWithValue(error)
+        }
+    
+})
+
+export const deleteUserPost=createAsyncThunk("post/deleteUserPost",async(postId,thunkAPI)=>{
+    try{
+        const token=localStorage.getItem("token");
+        const response=await deleteUserPostService(postId,token)
+        return response.data.posts;
+    }catch(error){
+return thunkAPI.rejectWithValue(error)
+    }
+})
+
+export const addUserPost=createAsyncThunk("post/addUserPost",async(postData,thunkAPI)=>{
+    try{
+        const token=localStorage.getItem("token");
+        const response=await addPostService(postData,token)
+        return response.data.posts;
+    }catch(error){
+        return thunkAPI.rejectWithValue(error)
+    }
+})
```

Comment:
Got an understanding about implementing async thunk API calls here!! Good work on this Shrushti!!